### PR TITLE
Remove array prototype extension

### PR DESCRIPTION
**Pull request by stopfstedt**
companion PR to https://github.com/ilios/common/pull/3079

- [x] `toArray()`
- [x] `sortBy()`
- [x] `filterBy()`
- [x] `findBy()`
- [x] `any()`
- [x] ~~`objectAt()`~~ n/a
- [x] `invoke()`
- [x] `mapBy()`
- [x] `uniq()`
- [x] `addObject()`
- [x] `pushObject()`
- [x] `pushObjects()`
- [x] `removeObject()`
- [x] `getEach()`
- [x] ~~`removeObjects()`~~ n/a
- [x] `firstObject`
- [x] `lastObject`

### CODE REVIEW

**Comment by dartajax** [comment]
I can test this a bit more tomorrow but so far it seems fine.

File: package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
     "ember-cli-server-variables": "3.0.0",
     "ember-cli-sri": "^2.1.1",
     "ember-cli-terser": "^4.0.2",
+    "ember-disable-prototype-extensions": "^1.1.3",
```
Comment (jrjohnson) [inline package.json:76]: I'm not sure if I should have gone this way in the addon, but for an app I think [following this guide](https://guides.emberjs.com/release/configuring-ember/disabling-prototype-extensions/) to disable the extensions is probably a better path.

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: app/components/global-search-box.js [hunk 2]
```diff
@@ -217,7 +218,7 @@ export default class GlobalSearchBox extends Component {
     yield timeout(DEBOUNCE_MS);
 
     const { autocomplete } = yield this.iliosSearch.forCurriculum(this.internalQuery, true);
-    this.autocompleteCache.pushObject({ q: this.internalQuery, autocomplete });
+    this.autocompleteCache.push({ q: this.internalQuery, autocomplete });
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply app/components/global-search-box.js:221]: :heavy_check_mark:

File: app/components/new-myreport.hbs [hunk 3]
```diff
@@ -135,7 +135,7 @@
                     <span class="details">
                       {{this.selectedMeshTerm.id}}
                       {{#if this.selectedMeshTerm.trees}}
-                        -{{this.selectedMeshTerm.trees.lastObject.treeNumber}}
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply app/components/new-myreport.hbs:138]: :heavy_check_mark:

File: app/components/new-myreport.hbs [hunk 4]
```diff
@@ -135,7 +135,7 @@
...
+                          {{get (object-at 0 (reverse this.selectedMeshTerm.trees)) "treeNumber"}}
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply app/components/new-myreport.hbs]: :heavy_check_mark:

File: package.json (hunk 1 above)
Comment (stopfstedt) [reply package.json]: :heavy_check_mark:

**Review by jrjohnson — APPROVED** [review]

**Review by dartajax — APPROVED** [review]
looks good to me functionally

**Pull request merged**